### DISCUSSION TITLE
Labels: Do we need a `Code Snippet` tag or is it covered by `Language Service`?

### DISCUSSION THREAD

**Discussion by Karmavil** [microsoft/vscode-cpptools#8454] [Q&A] [answered]
Hi.
I have an issue with the FOR code snippet (it vanished) and I would like to fix it or find a workaround.

Considering that this plugin covers the categories:
* Programming Languages
* **Snippets**
* Linters
* Debuggers
* Formatters

And considering also that I am not part of the maintainers but the users.
I would like to request the Tag.
It won't fix any issue neither create new ones (or maybe it does (both))

There are at least 2 issues related: 
[#8076](https://github.com/microsoft/vscode-cpptools/issues/8076) Feature request.
[#5919](https://github.com/microsoft/vscode-cpptools/issues/5919) Feature request.

Do you think it would help?
Have in mind there is also a User Snippet section in VS code so not sure but related to this plugin it has a lot of sense to me..
Thanks

**Comment by sean-mcmanus** [comment]
What exactly is the FOR snippet you're referring to? This one?
![image](https://user-images.githubusercontent.com/19859882/144113008-7bc77a41-5f32-45b8-902e-0149d699d601.png). I'm not reproing any "vanishing" behavior.

I don't know what you mean by "request the tag".

**Comment by Karmavil** [comment]
Hello @sean-mcmanus thanks for your answer.

#### The `tag`is for this repo, to filter issues (features requests, bugs, etc) related to Code Snippets

I didn't report any bug but now we're here I could provide some extra details..
The code snippet that behaves like if I did something wrong, or there is something misconfigured, is the one that you're showing in that screenshot.

Let me show you mines. 
Note: The pictures include the bottom of the pictures to show that vscode has some CMake configuration in the project where I'm having this problem.. I tried removing the .vscode folder but it didn't fix it.

![Screenshot from 2021-11-30 17-08-48](https://user-images.githubusercontent.com/2354060/144123773-9f625672-781c-43f5-9cc3-0341eaa6dc3f.png)

and below is a new folder / project I started yesterday so there is no doubts that my issue is related to that project

![Screenshot from 2021-11-30 17-09-09](https://user-images.githubusercontent.com/2354060/144123947-25ee2e57-859a-4ef1-ae98-a926066ad76e.png)

This is probably due to the configuration (meaning folder, workspace, or user configuration) but remember I started the discussion for the tag. Of course I would  appreciate any advice or observation but I didn't open a discussion for that.

**Reply by sean-mcmanus** [reply]
We use the "Feature: Auto-complete" to cover snippets -- the implementation for snippets is included as part of the auto-complete feature.

The bug where the "for" snippet is missing is very strange -- I don't see yet how that can be possible. It might be a VS Code bug, because we should be returning the snippet.

**Comment by Karmavil** [comment]
Thank you. I will investigate their repository

**Comment by Karmavil** [comment, marked as answer]
Well thank you very much. Good answer.
I'm closing the discussion given that already exists a `tag` for [filter those issues in the VS Code repository](https://github.com/microsoft/vscode/labels/snippets)

I'm not opening an issue for that because I don't know how to replicate the steps, besides it happens in that particular project / folder but it is gone if I clone the project in a new directory.

Thanks
